drivers/SmartThings/zigbee-window-treatment/src/aqara/roller-shade/init.lua

```diff
@@ -0,0 +1,159 @@
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local FrameCtrl = require "st.zigbee.zcl.frame_ctrl"
+local data_types = require "st.zigbee.data_types"
+local aqara_utils = require "aqara/aqara_utils"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+
+local initializedStateWithGuide = capabilities["stse.initializedStateWithGuide"]
+local reverseRollerShadeDirId = "stse.reverseRollerShadeDir"
+
+local shadeRotateState = capabilities["stse.shadeRotateState"]
+local shadeRotateStateId = "stse.shadeRotateState"
+local setRotateStateCommandName = "setRotateState"
+
+local MULTISTATE_CLUSTER_ID = 0x0013
+local MULTISTATE_ATTRIBUTE_ID = 0x0055
+local TILT_UP_VALUE = 0x0004
+local TILT_DOWN_VALUE = 0x0005
+
+local INITIALIZED_STATE = "initializedState"
+
+local function window_shade_level_cmd(driver, device, command)
+  -- Cannot be controlled if not initialized
+  local initialized = device:get_field(INITIALIZED_STATE) or 0
+  if initialized ~= 1 then
+    return
+  end
+
+  aqara_utils.shade_level_cmd(driver, device, command)
+end
+
+local function window_shade_open_cmd(driver, device, command)
+  -- Cannot be controlled if not initialized
+  local initialized = device:get_field(INITIALIZED_STATE) or 0

```

Comment:
you should be able to use `device:get_latest_state` for your custom capability rather than have this separate field

Reply:
e.g.:

```
local function window_shade_open_cmd(driver, device, command)
  local initialized = device:get_latest_state("main", initializedStateWithGuide.ID, initializedStateWithGuide.initializedStateWithGuide.NAME) or 0
  if initialized == initializedStateWithGuide.initializedStateWithGuide.initialized.NAME then
    device:send_to_component(...)
  end
end
```

something like that

having only one exit point for the code is easier to read as well

Reply:
Thanks. I'll update it.